### PR TITLE
fix: remove duplicated unit tests in JSONPatch and ApplyConfiguration

### PR DESCRIPTION
**Pull request by MariamFahmy98**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR removes the following duplicated tests:
1. `jsonPatch with false test operation` at file: staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/mutating/patch/json_patch_test.go
2. `apply configuration add to listType=map` at file: staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/mutating/patch/smd_test.go

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/mutating/patch/smd_test.go [hunk 1]
```diff
@@ -51,35 +51,6 @@ func TestApplyConfiguration(t *testing.T) {
...
-				},
-			}},
-			expectedResult: &appsv1.Deployment{Spec: appsv1.DeploymentSpec{
-				Template: corev1.PodTemplateSpec{
-					Spec: corev1.PodSpec{
-						Volumes: []corev1.Volume{{Name: "x"}, {Name: "y"}},
-					},
-				},
-			}},
-		},
```
Comment (cici37) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/mutating/patch/smd_test.go:82]: What is this test case duplicated to?

Comment (jpbetz) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/mutating/patch/smd_test.go:82]: It's directly below. I think this happened during a rebase.

**Pull request merged**